programs/survival/sleep_in.sc

```diff
@@ -0,0 +1,34 @@
+//sleep in by nsluhrs
+//right click a bed with a long night vision potion and the game time will skip to night
+
+global_night_time = 12550; 
+global_are_you_sure_night=false;
+__on_player_right_clicks_block(player, item_tuple, hand, block, face, hitvec) ->(
+  validClick=(item_tuple:2:'Potion'=='minecraft:long_night_vision' && block_tags(block, 'beds'));
+  if(validClick && (day_time()%24000)< global_night_time,
+    if(global_are_you_sure_night,
+      _skip_day();
+      global_are_you_sure_night=false;
+      
+    );
+    //prevents the prompt from showing up if you sleep right away
+    if(!global_are_you_sure_night && (day_time()%24000)< global_night_time-100,
+      print('are you sure you want to skip to night?');
+      print('click again to confirm');
+      global_are_you_sure_night = true;
+    )
+    );
+  //changes 
+  if(get(get(item_tuple,2),'Potion')!='minecraft:long_night_vision',
+  global_are_you_sure_night=false
+  )
+);
+
+_skip_day() -> (
+	//dtime = day_time()%24000; //day time
+	if((dtime = day_time()%24000)< global_night_time, // if it's sleepy time
+		current = day_time();
+		day_time(current + (global_night_time - dtime) );
+		print('day skipped')
+	);

```

Comment:
This dpoesn't work if it's already night. Again, if that's intended, disregard this, but I'm not sure that's intended.

Reply:
well I think it is intended, seen as though this script is trying to skip the day. If it's night, then there is no need to skip the day.